data/src/transform.py

```diff
@@ -224,8 +224,8 @@ def get_pod_metrics_from_cli():
     command = 'kubectl get pods -o wide --no-headers --all-namespaces'
     ret, out, err = shell_exec(command)
 
-    # 1:NS | 2:Name | 3:Ready | 4:Status | 5:Restarts | 6:Age | 7:IP | 8:Node
-    reg = re.compile(r"^([^ ]+)\s+([^ ]+)\s+([^ ]+)\s+([^ ]+)\s+([^ ]+)\s+([^ ]+)\s+([^ ]+)\s+([^ ]+)$")
+    # 1:NS | 2:Name | 3:Ready | 4:Status | 5:Restarts | 6:Age | 7:IP | 8:Node | 9: NOMINATED NODE
+    reg = re.compile(r"^([^ ]+)\s+([^ ]+)\s+([^ ]+)\s+([^ ]+)\s+([^ ]+)\s+([^ ]+)\s+([^ ]+)\s+([^ ]+)\s+([^ ]+)$")
```

Comment:
I would think that this breaks compatibility with K8s < v1.11.0. To make this more dynamic, the last group should be optional. However, then it would have to be wrapped into an outer optional capturing group (as the space `\s+`) must also be optional in case another element is coming.

I would suggest to make it more dynamic to switch over to named capturing groups:
```
^(?P<ns>[^ ]+)\s+(?P<name>[^ ]+)\s+(?P<ready>[^ ]+)\s+(?P<status>[^ ]+)\s+(?P<restarts>[^ ]+)\s+(?P<age>[^ ]+)\s+(?P<ip>[^ ]+)\s+(?P<node>[^ ]+)(\s+(?P<nnode>[^ ]+))?$
```
This is demonstrated here: https://regex101.com/r/3Suwbi/2

This would however also require code changes below.

Reply:
I suggest to use:

```
^([^\s]+)\s+([^\s]+)\s+([^\s]+)\s+([^\s]+)\s+([^\s]+)\s+([^\s]+)\s+([^\s]+)\s+([^\s]+)[^\n]*$
```

This way we keep the changes low. Later on code could be refactored to use the groups.

Reply:
:+1: